src/test/java/sysc4806/graduateAdmissions/controller/InterestControllerTest.java

```diff
@@ -0,0 +1,173 @@
+import sysc4806.graduateAdmissions.model.Interest;
+import sysc4806.graduateAdmissions.repositories.InterestRepository;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * Tests for the InterestController using MockMVC and a mocked InterestRepository
+ *
+ * @author luke
+ */
+@SpringBootTest
+@AutoConfigureMockMvc
+class InterestControllerTest {
+    private final MediaType APPLICATION_JSON_UTF8 =
+            new MediaType(MediaType.APPLICATION_JSON.getType(),
+                    MediaType.APPLICATION_JSON.getSubtype(),
+                    StandardCharsets.UTF_8);
+    private List<Interest> interests;
+    @Autowired
+    private MockMvc mockMvc;
+    @MockBean
+    private InterestRepository repo;
+
+    @BeforeEach
+    void setUpMocks() {
+        interests = Arrays.asList(
+                Interest.builder().id(0).department(Department.SYSC).keyword("cybersecurity").build(),
+                Interest.builder().id(1).department(Department.SYSC).keyword("testing").build(),
+                Interest.builder().id(2).department(Department.SYSC).keyword("machine_learning").build(),
+                Interest.builder().id(3).department(Department.MAAE).keyword("gears").build(),
+                Interest.builder().id(4).department(Department.SREE).keyword("grilled_cheese").build());
+
+        when(repo.findAll()).thenReturn(interests);
+        when(repo.findByDepartment(Department.SYSC)).thenReturn(interests.subList(0, 3));
+        when(repo.findByDepartment(Department.MAAE)).thenReturn(Collections.singletonList(interests.get(3)));
+        when(repo.findByDepartment(Department.SREE)).thenReturn(Collections.singletonList(interests.get(4)));
+        for(int i = 0; i <= 4; i++) {
+            when(repo.findById((long) i)).thenReturn(Optional.of(interests.get(i)));
+            doNothing().when(repo).deleteById((long) i);
+        }
+    }
+
+    /**
+     * helper method to convert objects into JSON format
+     *
+     * @param o the object to convert to JSON format
+     * @return a string contain the JSON for Object o
+     * @throws JsonProcessingException when JSON writing fails
+     */
+    private String toJson(Object o) throws JsonProcessingException {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(SerializationFeature.WRAP_ROOT_VALUE, false);
+        ObjectWriter ow = mapper.writer().withDefaultPrettyPrinter();
+        return ow.writeValueAsString(o);
+    }
```

Comment:
I would consider putting this in an abstract test parent or another test helper class for our other tests. I'm sure theres something we can import instead that will do the same thing though.

I wanna see this merged for Wednesday, so let's keep it as is. No change required